modules/react/text-input/lib/TextInput.tsx

```diff
@@ -21,74 +20,103 @@ export interface TextInputProps extends Themeable, GrowthBehavior {
+    '&:focus-visible:not([disabled]), &.focus:not([disabled])': {
+      borderColor: brand.common.focusOutline,
+      boxShadow: `inset 0 0 0 1px ${cssVar(brand.common.focusOutline)}`,
       outline: 'none',
     },
-    '&:disabled': {
-      backgroundColor: inputColors.disabled.background,
-      borderColor: inputColors.disabled.border,
-      color: inputColors.disabled.text,
+    '&:disabled, .disabled': {
+      backgroundColor: system.color.bg.alt.softer,
+      borderColor: system.color.border.input.disabled,
+      color: system.color.text.disabled,
       '&::placeholder': {
-        color: inputColors.disabled.text,
+        color: system.color.text.disabled,
       },
     },
-    '::-ms-clear': {
-      display: 'none',
-    },
-  },
-  ({width}) => ({
-    minWidth: width || 280,
-    width,
   }),
-  ({grow}) =>
-    grow && {
-      width: '100%',
+  modifiers: {
+    grow: {
+      true: {
+        width: '100%',
+        resize: 'vertical',
+      },
+      false: {
+        width: 'initial',
+      },
     },
-  ({theme, error}) => {
-    return {
-      '&:focus-visible:not([disabled]), &.focus:not([disabled])': {
-        borderColor: theme.canvas.palette.common.focusOutline,
-        boxShadow: `inset 0 0 0 1px ${theme.canvas.palette.common.focusOutline}`,
-        outline: 'none',
+
+    error: {
+      error: {
+        borderColor: brand.error.base,
+        boxShadow: `inset 0 0 0 ${px2rem(1)} ${brand.error.base}`,
+        '&:hover, &.hover, &:disabled, &.disabled, &:focus-visible:not([disabled]), &.focus:not([disabled])':
+          {
+            borderColor: brand.error.base,
+          },
+        '&:focus-visible:not([disabled]), &.focus:not([disabled])': {
+          boxShadow: `inset 0 0 0 ${px2rem(1)} ${brand.error.base}, 0 0 0 2px ${
+            system.color.border.inverse
+          }, 0 0 0 4px ${brand.common.focusOutline}`,
+        },
+      },
+      alert: {
+        borderColor: brand.alert.darkest,
+        boxShadow: `inset 0 0 0 ${px2rem(2)} ${brand.alert.base}`,
+        '&:hover, &.hover, &:disabled, &.disabled, &:focus-visible:not([disabled]), &.focus:not([disabled])':
+          {
+            borderColor: brand.alert.darkest,
+          },
+
+        '&:focus-visible:not([disabled]), &.focus:not([disabled])': {
+          boxShadow: `inset 0 0 0 ${px2rem(2)} ${brand.alert.base},
+        0 0 0 2px ${system.color.border.inverse},
+        0 0 0 4px ${brand.common.focusOutline}`,
+        },
       },
-      ...errorRing(error, theme),
-    };
-  }
-);
+    },
+  },
+  defaultModifiers: {
+    grow: 'false',
+  },
+});
 
 export const TextInput = createComponent('input')({
   displayName: 'TextInput',
   Component: ({grow, error, width, ...elemProps}: TextInputProps, ref, Element) => (
-    <StyledTextInput
+    <Element
       type="text"
       ref={ref}
-      as={Element}
-      grow={grow}
-      error={error}
-      width={width}
-      {...elemProps}
+      {...mergeStyles(
```

Comment:
I didn't see style props handled before. I think we should only use `mergeStyles` if the component originally used `Box`, `Flex`, or `Grid` layout components. Otherwise, this changes the prop interface.

```suggestion
      {...handleCsProp(
```